src/commands/remoteDebug/remoteDebugCommon.ts

```diff
@@ -57,19 +57,25 @@ function isNodeVersionSupported(nodeVersion: string): boolean {
     return (major > 8 || (major === 8 && minor >= 11));
 }
 
-export async function setRemoteDebug(isRemoteDebuggingToBeEnabled: boolean, confirmMessage: string, noopMessage: string | undefined, siteClient: SiteClient, siteConfig: SiteConfigResource, progress: vscode.Progress<{}>): Promise<void> {
+export async function setRemoteDebug(isRemoteDebuggingToBeEnabled: boolean, confirmMessage: string, noopMessage: string | undefined, siteClient: SiteClient, siteConfig: SiteConfigResource, progress?: vscode.Progress<{}>): Promise<void> {
     if (isRemoteDebuggingToBeEnabled !== siteConfig.remoteDebuggingEnabled) {
         const result: vscode.MessageItem = await ext.ui.showWarningMessage(confirmMessage, { modal: true }, DialogResponses.yes, DialogResponses.learnMore, DialogResponses.cancel);
         if (result === DialogResponses.yes) {
             siteConfig.remoteDebuggingEnabled = isRemoteDebuggingToBeEnabled;
+            if (progress) {
+                reportMessage('Updating site configuration to set remote debugging...', progress);
+            }
 
-            reportMessage('Updating site configuration to set remote debugging...', progress);
             await callWithTelemetryAndErrorHandling('appService.remoteDebugUpdateConfiguration', async function (this: IActionContext): Promise<void> {
                 this.suppressErrorDisplay = true;
                 this.rethrowError = true;
                 await siteClient.updateConfiguration(siteConfig);
             });
-            reportMessage('Updating site configuration done...', progress);
+
+            if (progress) {
+                reportMessage('Updating site configuration done...', progress);
+            }
+
         } else if (result === DialogResponses.learnMore) {
             await openUrl('https://aka.ms/appsvc-remotedebug');
```

Comment:
This learn more link won't make sense in your ssh scenario. I'm fine with creating your own learn more link or just removing the learn more button in this case.

It might make your life easier to use the shared `learnMoreLink` support I added a little while ago. (Where you just pass in learnMoreLink in the options to `ext.ui.showWarningMessage` and it will handle adding the learn more button if that link is defined). Like [here](https://github.com/Microsoft/vscode-azurefunctions/blob/4810e0ebaec0b1ad78d820ad0e519d3227019dd2/src/commands/createNewProject/verifyIsProject.ts#L60)